app/lib/actions/pulp/consumer/abstract_content_action.rb

```diff
@@ -13,6 +13,21 @@ def external_task=(external_task_data)
           end
         end
 
+        def process_timeout
+          accept_timeout = ::Katello.config.content_action_accept_timeout
+          finish_timeout = ::Katello.config.content_action_finish_timeout
+          pulp_state = output[:pulp_tasks][0][:state]
+
+          if pulp_state == 'waiting'
+            fail _("Host did not respond within %s seconds.  Is katello-agent installed and goferd running on the Host?") % accept_timeout
+          elsif output[:client_accepted].nil?
```

Comment:
Because I need to check the first condition first.  This method may be called up to 2 times.

First time, if the client hasn't picked up the task, throw an error.  If it has picked up the task, save the client_accepted time and set another timer. 
Second time (after the 2nd timer is triggered), fail if the state is still running.